### ISSUE TITLE
Prompt length is incorrectly calculated when using unicode characters

### ISSUE THREAD

**Issue by BYK** [scopatz/xonsh#631]
I wanted to make Xonsh prompt look more like powerline and used some unicode characters in it. When I did this, the cursor was places a few characters ahead of the prompt end. This becomes quite annoying once you have more than 2 unicode characters.

This may be a PTK issue but I was unable to trace it properly in limited time.

**Comment by gforsyth** [comment]
Hey @BYK -- can you post the `$PROMPT` string you used?  I'm can't replicate this at the moment, but I'm just using random unicode characters.

**Comment by scopatz** [comment]
Also, @BYK, can you post your encoding or the results of the `xonfig` command, please? For what it is worth, I think that this is a ptk problem.

**Comment by BYK** [comment]
Oh wow, something fixed this. I don't think it was PTK because I tried upgrading it multiple times without any luck. I'm guessing some recent refactoring in Xonsh fixed it. I was using `\u26D5` and `\u25B6` characters (first as the branch symbol and second for the path separator). 

I was using UTF-8 encoding in my terminal.

@scopatz `xonfig` expects an argument I guess since it says this:
```
BYK@ ~▶Documents▶Projects▶xonsh⛕colors$ xonfig
TypeError: bad argument type for built-in operation
BYK@ ~▶Documents▶Projects▶xonsh⛕colors$
```

**Comment by scopatz** [comment]
Awesome @BYK!  Glad that is it working.  On the `xonfig` issue, can you post that with a full traceback, please? It is hard to see where the error is coming from. Thanks.

**Comment by BYK** [comment]
@scopatz - Oh, right sorry. I forgot to turn traceback on. This is what I got:
```
$ xonfig
Traceback (most recent call last):
  File "C:\Program Files\Python\lib\site-packages\xonsh\proc.py", line 338, in wrapped_simple_command
    r = f(args, i)
  File "C:\Program Files\Python\lib\site-packages\xonsh\aliases.py", line 168, in xonfig
    return main(args)
  File "C:\Program Files\Python\lib\site-packages\xonsh\xonfig.py", line 285, in main
    return _MAIN_ACTIONS[ns.action](ns)
  File "C:\Program Files\Python\lib\site-packages\xonsh\xonfig.py", line 237, in _info
    ('have readline', is_readline_available()),
  File "C:\Program Files\Python\lib\site-packages\xonsh\shell.py", line 15, in is_readline_available
    import readline
  File "C:\Program Files\Python\lib\site-packages\readline.py", line 6, in <module>
    from pyreadline.rlmain import Readline
  File "C:\Program Files\Python\lib\site-packages\pyreadline\__init__.py", line 11, in <module>
    from . import unicode_helper, logger, clipboard, lineeditor, modes, console
  File "C:\Program Files\Python\lib\site-packages\pyreadline\console\__init__.py", line 15, in <module>
    from .console import *
  File "C:\Program Files\Python\lib\site-packages\pyreadline\console\console.py", line 21, in <module>
    import pyreadline.unicode_helper as unicode_helper
AttributeError: module 'pyreadline' has no attribute 'unicode_helper'
```
Upgrading `pyreadline` solved the issue but this was a bit unexpected since I was using PTK and didn't expect `pyreadline` to be a required dependency anymore.

FWIW my encoding was reported as `utf-8` :)

**Comment by scopatz** [comment]
Thanks @BYK, I have put in a fix in #636.  This was a problem internal to pyreadline, which is why updating probably fixed it.
